### ISSUE TITLE
Learning does not start

### ISSUE THREAD

**Issue by FiReTiTi** [fchollet/keras#1926]
I wanted to reproduce the issues that I cited in [this post](https://github.com/fchollet/keras/issues/1917).
So I wanted to train a neural network using keras, but the training does not start, and it happens in really weird situations.

Here is a portion of my code:

	model = Sequential()
	model.add(Convolution2D(8, 7, 7, border_mode='valid', input_shape=(1, 31, 31), activation='tanh'))
	model.add(MaxPooling2D(pool_size=(2, 2)))
	model.add(Convolution2D(16, 5, 5, border_mode='valid', activation='tanh'))
	model.add(MaxPooling2D(pool_size=(2, 2)))
	model.add(Convolution2D(32, 3, 3, border_mode='valid', activation='relu'))
	model.add(Flatten())
	model.add(Dense(23))
	model.add(Activation('tanh'))
	model.add(Dropout(0.1))
	model.add(Dense(11))
	model.add(Activation('sigmoid'))
	model.add(Dropout(0.1))
	model.add(Dense(1))
	model.add(Activation('sigmoid'))

	optimizer = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
	print("optimizer done")
	model.compile(loss='binary_crossentropy', optimizer=optimizer)
	print("compile done")
	sys.stdout.flush()

	model.fit(dataset, labels, batch_size=batch_size, nb_epoch=nb_epoch, shuffle=True, validation_split=0.1, verbose=2)

If I run this code in the terminal using `python mynn.py`, everything goes fine and the training starts.

But if I run this code in the terminal using `python mynn.py &`, `python mynn.py > results.txt`, or if I redirect the outputs in my code `sys.stdout = open('Outputs.txt', 'w')`, then the program stall for the `fit` method, so the learning does not start. The `compile done`is displayed, but nothing more, no training.

I took a look and it seems that the program keeps calling this function `clock_gettime(CLOCK_MONOTONIC_RAW, {,}) = 0`

Any ideas why it does not start when I redirect the outputs?

**Comment by FiReTiTi** [comment]
Ok, I got the answer of that.
The training starts, but all the outputs are displayed/written only when the training is over, nothing before.

**Comment by philipperemy** [comment]
@FiReTiTi can you close this case then?
